genie-web/src/test/groovy/com/netflix/genie/web/tasks/job/JobCompletionServiceSpec.groovy

```diff
@@ -224,4 +261,50 @@ class JobCompletionServiceSpec extends Specification {
             assert !d.exists()
         })
     }
+
+    /**
+     * Verify the presence of failure tags
+     *
+     * @param tags tags to be verified
+     * @param throwable exception thrown on failure
+     * @return true if all failure tags exist
+     */
+    private Boolean verifyFailureTags(Map<String, String> tags,
+                                      final Throwable throwable) {
+        final Map<String, String> failureTags = MetricsUtils.newFailureTagsMapForException(throwable);
+
+        return tags.get(MetricsConstants.TagKeys.STATUS)
+                .equals(failureTags.get(MetricsConstants.TagKeys.STATUS)) \
+         && tags.get(MetricsConstants.TagKeys.EXCEPTION_CLASS)
+                .equals(failureTags.get(MetricsConstants.TagKeys.EXCEPTION_CLASS)) \
+         && verifyCommonTags(tags);
+    }
+
+    /**
+     * Verify the presence of success tags
+     *
+     * @param tags tags to be verified
+     * @return true if all success tags exist
+     */
+    private Boolean verifySuccessTags(final Map<String, String> tags) {
+        final Map<String, String> successTags = MetricsUtils.newSuccessTagsMap();
+
+        return tags.get(MetricsConstants.TagKeys.STATUS).
+                equals(successTags.get(MetricsConstants.TagKeys.STATUS)) \
+                 && verifyCommonTags(tags);
+    }
+
+    /**
+     * Verify the presence of common tags
+     *
+     * @param tags tags to be verified
+     * @return true if common tags exist
+     */
+    private Boolean verifyCommonTags(final Map<String, String> tags) {
+        return tags.containsKey(MetricsConstants.TagKeys.COMMAND_ID) \
```

Comment:
Since you are using the same values for IDs, you should check the values in this map are what you expect, not just that the keys are present.